src/modules/Dashboard/features/Applet/Schedule/ImportSchedulePopup/ImportSchedulePopup.utils.test.tsx

```diff
@@ -0,0 +1,415 @@
+    expect(result).toBeNull();
+  });
+
+  test.each`
+    type                                        | expectedText
+    ${ImportScheduleErrors.StartTime}           | ${'Activity Start Time. Valid data format: HH:mm for Scheduled activity and - for Always available activity.'}
+    ${ImportScheduleErrors.EndTime}             | ${'Activity End Time. Valid data format: HH:mm for Scheduled activity and - for Always available activity.'}
+    ${ImportScheduleErrors.NotificationTime}    | ${'Notification Time. Valid data format: HH:mm or -.'}
+    ${ImportScheduleErrors.Frequency}           | ${'Frequency. Valid data format: Always, Once, Daily, Weekly, Monthly, Weekdays.'}
+    ${ImportScheduleErrors.Date}                | ${'Date. Valid data format: DD Month YYYY.'}
+    ${ImportScheduleErrors.StartEndTime}        | ${'Activity End Time should not be equal to Activity Start Time.'}
+    ${ImportScheduleErrors.BetweenStartEndTime} | ${'Notification Time should be between Activity Start Time and Activity End Time.'}
+  `('renders error text for type=$type', ({ type, expectedText }) => {
+    const { container } = renderWithProviders(getInvalidError(type));
+    expect(container).toHaveTextContent(expectedText);
+  });
+});
+
+describe('getUploadedDate', () => {
+  test.each`
+    inputDate                 | expectedOutput
+    ${'01 January 2022'}      | ${new Date('2022-01-01')}
+    ${'15-Feb-2023'}          | ${new Date('2023-02-15')}
+    ${new Date('2021-12-25')} | ${new Date('2021-12-25')}
+    ${'invalidDateString'}    | ${'invalidDateString'}
+  `('returns $expectedOutput for input $inputDate', ({ inputDate, expectedOutput }) => {
+    const result = getUploadedDate(inputDate);
+    expect(result).toEqual(expectedOutput);
+  });
+
+  test.each`
+    inputDate                 | isValid
+    ${'01 January 2022'}      | ${true}
+    ${'15-Feb-2023'}          | ${true}
+    ${new Date('2021-12-25')} | ${false}
+    ${'invalidDateString'}    | ${false}
+  `('validates date format correctly for input $inputDate', ({ inputDate, isValid }) => {
+    const isValidDate = dateValidationRegex.test(inputDate);
+    expect(isValidDate).toBe(isValid);
+  });
+});
+
+describe('getUploadedTime', () => {
+  test.each`
+    inputTime        | expectedOutput
+    ${'12:30'}       | ${'12:30'}
+    ${'3:45'}        | ${'03:45'}
+    ${'invalidTime'} | ${'invalidTime'}
+    ${new Date()}    | ${format(new Date(), DateFormats.Time)}
+  `('returns $expectedOutput for input $inputTime', ({ inputTime, expectedOutput }) => {
+    const result = getUploadedTime(inputTime);
+    expect(result).toEqual(expectedOutput);
+  });
+
+  test('replaces single-digit hour with leading zero', () => {
+    const result = getUploadedTime('1:15');
+    expect(result).toBe('01:15');
+  });
+});
+
+describe('getEndOfYearDate', () => {
+  const testCases = [
+    {
+      uploadedDate: new Date('2022-03-05'),
+      expected: endOfYear(new Date()),
```

Comment:
please check if tests with new Date will not be failed in different dates in future. I guess and maybe right solution to be safe when testing dates is to use [jest.setSystemTime](https://jestjs.io/docs/jest-object#jestsetsystemtimenow-number--date)